admyral/compiler/action_parser.py

```diff
@@ -218,18 +222,26 @@ def _parse_type_annotation(arg: ast.arg) -> tuple[str, bool, dict[str, str]]:
             "Argument metadata must be provided using ArgumentMetadata(...)."
         )
 
-    arg_metadata_parmas = {}
+    # get type and check whether it is optional
+    arg_type = astor.to_source(arg_type_ast).strip()  # TODO: improve this
+    is_optional = is_optional_type(arg_type_ast)
+
+    arg_metadata_params = {}
     for keyword in argument_metadata_ast.keywords:
         if not isinstance(keyword.value, ast.Constant) or not isinstance(
             keyword.value.value, str
         ):
             raise ValueError("Argument metadata parameters must be a string.")
-        if keyword.arg in arg_metadata_parmas:
+        if keyword.arg in arg_metadata_params:
             raise ValueError(
                 f"Found duplicate ArgumentMetadata parameter: {keyword.arg}. ArgumentMetadata parameters must be unique."
             )
-        arg_metadata_parmas[keyword.arg] = keyword.value.value
-    arg_metadata = ArgumentMetadata.model_validate(arg_metadata_parmas)
+        arg_metadata_params[keyword.arg] = keyword.value.value
+
+    if not arg_metadata_params:
+        raise ValueError("ArgumentMetadata must contain at least one parameter.")
```

Comment:
This check is not needed imo because this is already handled by `ArgumentMetadata.model_validate`